terraform/main.tf

```diff
@@ -45,6 +45,14 @@ module "network" {
   kv_certificate_subject                    = var.kv_certificate_subject
 }
 
+# Create App Insights
+module "appinsights" {
```

Comment:
Does this need to be it's own module, as opposed to being a part of webapp?

Reply:
It doesn't have to be technically, but to use appInsights we have to set some app settings which are set in the locals file at the main root. This module outputs the app setting values and then I've updated the locals to keep them all in one place instead of having some appsettings in local and others directly on the app service. Happy to discuss though :-)

Reply:
I see what you mean. The reason I ask is because the other modules have been used to break the configuration into distinct logical components e.g. web app, database, network etc. This is actually their recommended pattern -https://developer.hashicorp.com/terraform/tutorials/modules/pattern-module-creation. For me, App Insights doesn't best fit as it's own module given it's a fairly granular piece of configuration as opposed to a logical component.

So, the choice is whether to create a module for App Insights or to merge the appsettings in the configuration. I'd say the latter is probably the more favourable option as App Insights can be considered a part of the overall web app module.

Reply:
That makes sense. I've refactored it now to add the AppInsights into the web-app and used the merge function for the app settings. Cheers Sunny